src/lfortran/semantics/ast_body_visitor.cpp

```diff
@@ -2867,7 +2872,12 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
             ADD_ASR_DEPENDENCIES(current_scope, final_sym, current_function_dependencies);
         }
         ASRUtils::insert_module_dependency(final_sym, al, current_module_dependencies);
-        if( f ) {
+        if (f) {
+            int offset = (v_expr != nullptr) && (!nopass);
```

Comment:
Can you rewrite this using if statements --- I don't understand what it is doing. Is this something like:
```suggestion
            int offset = 0;
            if ((v_expr != nullptr) && (!nopass)) offset = 1;
```

Reply:
Possibly even clearer:
```suggestion
            int offset;
            if (v_expr == nullptr || nopass) {
                offset = 0;
            } else {
                offset = 1;
            }
```